File: api-dotnet/src/Application/Features/Scans/DTOs/ScanJob.cs
namespace Application.Features.Scans.DTOs;

public sealed record ScanJob(
    string DomainId,
    string DomainName,
    string RepoId,
    string ScanId,
    string ScanType,
    List<string> SurfaceTypes,
    string RequestedBy,
    string EnqueuedAt
);

File: worker-java/src/main/java/com/vulnwatch/worker/model/ScanJob.java
package com.vulnwatch.worker.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record ScanJob(
        @JsonProperty("DomainId") String domainId,
        @JsonProperty("DomainName") String domainName,
        @JsonProperty("RepoId") String repoId,
        @JsonProperty("ScanId") String scanId,
        @JsonProperty("ScanType") String scanType,
        @JsonProperty("SurfaceTypes") List<String> surfaceTypes,
        @JsonProperty("RequestedBy") String requestedBy,
        @JsonProperty("EnqueuedAt") String enqueuedAt
) {}